$NetBSD: patch-CMakeLists.txt,v 1.3 2026/01/04 23:29:54 gdt Exp $

Don't have a file called "version", to avoid conflict with C++20 <version>.

CMake will detect the MacOS 6-argument sendfile() support and configure
HAVE_SENDFILE, but the code itself doesn't support this interface.  So don't
probe for HAVE_SENDFILE if we HAVE_SENDFILE6_SUPPORT.

Just because librt doesn't exist does not mean that clock_gettime() doesn't
exist, so don't insist on librt being present.  Fixes the build on MacOS.

Let Pkgsrc be responsible for zm.conf into examples, rather than overwriting
the existing configuration file.

See ../files/upstream-PR-3914.

--- CMakeLists.txt.orig	2025-12-07 16:18:50.000000000 +0000
+++ CMakeLists.txt
@@ -4,7 +4,7 @@
 #
 cmake_minimum_required(VERSION 3.5.0)
 project(zoneminder)
-file(STRINGS "version" zoneminder_VERSION)
+file(STRINGS "version.txt" zoneminder_VERSION)
 # make API version a minor of ZM version
 set(zoneminder_API_VERSION "${zoneminder_VERSION}.1")
 
@@ -119,14 +119,16 @@ set(ZM_CONTENTDIR "/var/lib/zoneminder" 
 set(ZM_FONTDIR "${CMAKE_INSTALL_FULL_DATADIR}/zoneminder/fonts" CACHE PATH
   "Location of the font files used for timestamping, default: <prefix>/${CMAKE_INSTALL_DATADIR}/zoneminder/fonts")
 
+set(ZM_DB_TYPE "mysql" CACHE STRING
+  "Type of database where the ZoneMinder database located, default: mysql")
 set(ZM_DB_HOST "localhost" CACHE STRING
-  "Hostname where ZoneMinder database located, default: localhost")
+  "Hostname where the ZoneMinder database located, default: localhost")
 set(ZM_DB_NAME "zm" CACHE STRING
-  "Name of ZoneMinder database, default: zm")
+  "Name of the ZoneMinder database, default: zm")
 set(ZM_DB_USER "zmuser" CACHE STRING
-  "Name of ZoneMinder database user, default: zmuser")
+  "Name of the ZoneMinder database user, default: zmuser")
 set(ZM_DB_PASS "zmpass" CACHE STRING
-  "Password of ZoneMinder database user, default: zmpass")
+  "Password of the ZoneMinder database user, default: zmpass")
 set(ZM_WEB_USER "" CACHE STRING
   "The user apache or the local web server runs on. Leave empty for automatic detection.
      If that fails, you can use this variable to force")
@@ -269,7 +271,9 @@ check_include_file("ucontext.h" HAVE_UCO
 check_include_file("sys/sendfile.h" HAVE_SYS_SENDFILE_H)
 check_include_file("sys/syscall.h" HAVE_SYS_SYSCALL_H)
 check_function_exists("syscall" HAVE_SYSCALL)
+if(NOT HAVE_SENDFILE6_SUPPORT)
 check_function_exists("sendfile" HAVE_SENDFILE)
+endif(NOT HAVE_SENDFILE6_SUPPORT)
 check_function_exists("posix_memalign" HAVE_POSIX_MEMALIGN)
 check_type_size("siginfo_t" HAVE_SIGINFO_T)
 check_type_size("ucontext_t" HAVE_UCONTEXT_T)
@@ -280,7 +284,8 @@ if(UNIX)
   include(CheckLibraryExists)
   CHECK_LIBRARY_EXISTS(rt clock_gettime "time.h" HAVE_CLOCK_GETTIME)
   if(NOT HAVE_CLOCK_GETTIME)
-    message(FATAL_ERROR "clock_gettime not found")
+    # \todo Explain why just skipping this is ok.
+    #message(FATAL_ERROR "clock_gettime not found")
   else()
     list(APPEND ZM_BIN_LIBS "-lrt")
   endif()
@@ -810,7 +815,6 @@ set(WEB_PREFIX "${ZM_WEBDIR}")
 set(CGI_PREFIX "${ZM_CGIDIR}")
 set(WEB_USER "${ZM_WEB_USER}")
 set(WEB_GROUP "${ZM_WEB_GROUP}")
-set(ZM_DB_TYPE "mysql")
 if(ZM_PERL_SEARCH_PATH)
   set(EXTRA_PERL_LIB "use lib '${ZM_PERL_SEARCH_PATH}'; # Include custom perl install path")
 else()
@@ -873,8 +877,8 @@ else()
 endif()
 
 # Install zm.conf
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/zm.conf" DESTINATION "${ZM_CONFIG_DIR}")
-install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/conf.d/" DESTINATION "${ZM_CONFIG_SUBDIR}" PATTERN "*.in" EXCLUDE)
+#install(FILES "${CMAKE_CURRENT_BINARY_DIR}/zm.conf" DESTINATION "${ZM_CONFIG_DIR}")
+#install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/conf.d/" DESTINATION "${ZM_CONFIG_SUBDIR}" PATTERN "*.in" EXCLUDE)
 
 # Uninstall target
 configure_file(
